rotkehlchen/exchanges/binance.py

```diff
@@ -712,6 +748,98 @@ def query_lending_interests_history(
+                        f'Ignoring its lending interest history query.',
+                    )
+                    continue
+                except DeserializationError as e:
+                    self.msg_aggregator.add_error(
+                        f'Error at deserializing {self.name} asset. {e!s}. '
+                        f'Ignoring its lending interest history query.',
+                    )
+                    continue
+
+                try:
+                    asset = asset_from_binance(entry['asset'])
+                except (UnsupportedAsset, UnknownAsset) as e:
+                    error_type = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                    self.msg_aggregator.add_warning(
+                        f'Found {error_type} {self.name} asset {e.identifier}. '
+                        f'Ignoring its lending interest history query.',
+                    )
+                    continue
+
+                try:
+                    usd_price = PriceHistorian.query_historical_price(
+                        from_asset=asset,
+                        to_asset=A_USD,
+                        timestamp=ts_ms_to_sec(timestamp),
+                    )
+                    usd_value = usd_price * interest_received
+                except NoPriceForGivenTimestamp as e:
+                    log.warning(
+                        f'Could not find USD price of {asset} at {timestamp}. {e!s} '
+                        f'Using zero usd_value for lending history entry.',
+                    )
+                    usd_value = ZERO
+
+                event = HistoryEvent(
+                    event_identifier=hashlib.sha256(str(entry).encode()).hexdigest(),  # entry hash
+                    sequence_index=0,  # since event_identifier is always different
+                    timestamp=timestamp,
+                    location=self.location,
+                    location_label=self.name,  # the name of the CEX instance
+                    asset=asset,
+                    balance=Balance(
+                        amount=interest_received,
+                        usd_value=usd_value,
+                    ),
+                    notes=notes,
+                    event_type=HistoryEventType.RECEIVE,
+                    event_subtype=HistoryEventSubType.REWARD,
+                )
+                events.append(event)
+
+            if len(events) != 0:
+                history_events_db = DBHistoryEvents(self.db)
+                with self.db.user_write() as write_cursor:
+                    try:
+                        history_events_db.add_history_events(write_cursor, events)
+                    except InputError as e:
+                        self.msg_aggregator.add_error(
+                            f'Failed to save Binance {self.name} lending interest history from '
+                            f'{query_start_ts} to {query_end_ts} in the database. {e!s}',
+                        )
```

Comment:
Use the single one. The one you use is just a for loop. If for some reason an event is requeried with the change you made it won't be duplicated in the DB due to the event identifier but would fail with an sqlcipher Integrity error. Also why do you catch InputError. Where did you see this?

The probable exceptions according to the code are:
```
        - DeserializationError if the event could not be serialized for the DB
        - sqlcipher.IntegrityError: If the asset of the added history event does not exist in
        the DB. Can only happen if an event with an unresolved asset is passed.
```

```suggestion
                    for event in events:
                    try:
                        history_events_db.add_history_event(write_cursor, event)
                    except (IntegrityError, DeserializationError) as e:
                        log.error(f'Did not add lending history event {event} to the DB due to {e})
```